ETWAnalyzer_uTest/EventDump/DumpTcpTests.cs

```diff
@@ -0,0 +1,112 @@
+                DumpCommand dump = (DumpCommand)CommandFactory.CreateCommand(args);
+                dump.Parse();
+                dump.Run();
+                DumpTcp tcpDumper = (DumpTcp)dump.myCurrentDumper;
+
+                Assert.Equal(input.Item3.Item1, tcpDumper.MinMaxConnectionDurationS.Min);
+                Assert.Equal(input.Item3.Item2, tcpDumper.MinMaxConnectionDurationS.Max);
+
+            }
+
+        }
+
+        public List<string> Messages { get; set; } = new List<string>();
+
+        private ITestOutputHelper myWriter;
+
+        public DumpTcpTests(ITestOutputHelper myWriter)
+        {
+            this.myWriter = myWriter;
+        }
+
+        internal void Add(string message)
+        {
+            Messages.Add(message);
+        }
+
+        static readonly SocketConnection ipRemPort = new("123.205.91.10", 10000);
+        static readonly SocketConnection ipSrcPort = new("123.456.789.10", 100);
+
+        static readonly DateTimeOffset timeStampOpen = new DateTimeOffset(2023, 8, 11, 10, 05, 00, TimeSpan.Zero);
+        static readonly DateTimeOffset timeStampClose = new DateTimeOffset(2023, 8, 11, 10, 05, 00, TimeSpan.Zero);
+        static readonly DateTimeOffset timeStampClose1 = new DateTimeOffset(2023, 8, 11, 10, 05, 10, TimeSpan.Zero);
+        static readonly DateTimeOffset timeStampClose2 = new DateTimeOffset(2023, 8, 11, 10, 05, 20, TimeSpan.Zero);
+        static readonly TcpConnection tcpConnection = new(123, ipRemPort, ipSrcPort, timeStampOpen, timeStampClose, "", 100, 120, 100, 100, ETWProcessIndex.Invalid);
+        static readonly TcpConnection tcpConnection1 = new(456, ipRemPort, ipSrcPort, timeStampOpen, timeStampClose1, "", 100, 120, 100, 100, ETWProcessIndex.Invalid);
+        static readonly TcpConnection tcpConnection2 = new(789, ipRemPort, ipSrcPort, timeStampOpen, timeStampClose2, "", 100, 120, 100, 100, ETWProcessIndex.Invalid);
+        
+        
+
+        List<DumpTcp.MatchData> CreateTestData()
+        {
+            List<DumpTcp.MatchData> data = new List<DumpTcp.MatchData>
+            { 
+                new DumpTcp.MatchData
+                {
+                    Connection = tcpConnection,
+                },
+                new DumpTcp.MatchData
+                {
+                    Connection = tcpConnection1,
+                },
+                new DumpTcp.MatchData
+                {
+                    Connection = tcpConnection2,
+                }
+            };
+
+            return data;
+        }
+
+        [Fact]
+        public void MinMaxConnectionDurationFilterTest()
+        {
+            DumpTcp tcpDumper = new();
+            
+            var data = CreateTestData();
+         
+            tcpDumper.MinMaxConnectionDurationS = new MinMaxRange<double>(0, 10);
+
+            Assert.True(tcpDumper.MinMaxConnectionDurationFilter(data[0].Connection.TimeStampOpen, data[0].Connection.TimeStampClose));
```

Comment:
the passed data can be directly added to the method. It would be better to declare the timestamps and connections inside the method with descriptive names. Currently it is hard to see if the MinMaxConnectionDurationFilter works correctly. 

internal bool MinMaxConnectionDurationFilter(DateTimeOffset? connectTime, DateTimeOffset? closeTime) expects only primitive types.

You can therefore do 
DateTimeOffset connect_0 = new DateTimeOffset(2000,1,1,0,0,0,TimeSpan.Zero);
DateTimeOoffset connect_5s = connect_0 + TimeSpan.FromSeconds(5);
DateTimeOfffset connect_10s= connect_0 + TimeSpan.FromSeconds(10); 
DateTimeOfffset connect_11s= connect_0 + TimeSpan.FromSeconds(11); 

Then you can write it like this: 
   Assert.IsTrue( tcpDumper.MinMaxConnectionDurationFilter(connect_0, connect_5s) );
Then it is much more obvious which timestamps are tested. 
Then you do not need CreateTestData